### PR TITLE
Feature extractor fit

### PR DESCRIPTION
**Pull request by piotrekp1**
I refactored feature-extractor and therefore fixed all the issues that landscape had with it, I also noticed a little bug in featureextractor. List with points which are ready to compare was starting one point "too late". It didn't include the first point. I fixed it, and so i had to change the value in classifier test a little, that's why test_classifier.py was edited.

if you guys want to see exactly what I am talking about i created a simple test named test_feature_extractor which checks it, and if you want to see exact values, just add print(feature) so that code looks something like that:

 ```

   for feature in features:
        print(feature)
        if feature < minimum:
```

### CODE REVIEW

**Comment by michal7352** [comment]
I found bug:
Think of a symbol you want the application to learn and draw it 6 times.

New portion of events:
Event # 1 	out of 2 in the list. x: 821 y: 843
Event # 2 	out of 2 in the list. x: -1 y: -1

training...
ok


New portion of events:
Event # 1 	out of 3 in the list. x: 806 y: 852
Event # 2 	out of 3 in the list. x: 803 y: 846
Event # 3 	out of 3 in the list. x: -1 y: -1

training...
ok


New portion of events:
Event # 1 	out of 16 in the list. x: 902 y: 594
Event # 2 	out of 16 in the list. x: 893 y: 615
Event # 3 	out of 16 in the list. x: 864 y: 676
Event # 4 	out of 16 in the list. x: 821 y: 752
Event # 5 	out of 16 in the list. x: 795 y: 818
Event # 6 	out of 16 in the list. x: 790 y: 863
Event # 7 	out of 16 in the list. x: 810 y: 898
Event # 8 	out of 16 in the list. x: 853 y: 924
Event # 9 	out of 16 in the list. x: 942 y: 944
Event # 10 	out of 16 in the list. x: 1058 y: 949
...

training...
ok


New portion of events:
Event # 1 	out of 3 in the list. x: 970 y: 693
Event # 2 	out of 3 in the list. x: 969 y: 691
Event # 3 	out of 3 in the list. x: -1 y: -1

training...
ok


New portion of events:
Event # 1 	out of 59 in the list. x: 2096 y: 859
Event # 2 	out of 59 in the list. x: 2060 y: 851
Event # 3 	out of 59 in the list. x: 1970 y: 833
Event # 4 	out of 59 in the list. x: 1812 y: 803
Event # 5 	out of 59 in the list. x: 1641 y: 763
Event # 6 	out of 59 in the list. x: 1438 y: 711
Event # 7 	out of 59 in the list. x: 1220 y: 677
Event # 8 	out of 59 in the list. x: 1040 y: 660
Event # 9 	out of 59 in the list. x: 806 y: 658
Event # 10 	out of 59 in the list. x: 745 y: 666
...

training...
ok


New portion of events:
Event # 1 	out of 64 in the list. x: 891 y: 735
Event # 2 	out of 64 in the list. x: 896 y: 734
Event # 3 	out of 64 in the list. x: 907 y: 732
Event # 4 	out of 64 in the list. x: 925 y: 727
Event # 5 	out of 64 in the list. x: 951 y: 720
Event # 6 	out of 64 in the list. x: 984 y: 719
Event # 7 	out of 64 in the list. x: 1018 y: 718
Event # 8 	out of 64 in the list. x: -1 y: -1
Event # 9 	out of 64 in the list. x: 938 y: 675
Event # 10 	out of 64 in the list. x: 942 y: 680
...

training...
ok
learning...
Traceback (most recent call last):
  File "app.py", line 86, in <module>
    main()
  File "app.py", line 84, in main
    system_bitness)
  File "/home/michal/io-project/io-touchpad/app/threads/application.py", line 60, in application_thread
    classifier)
  File "/home/michal/io-project/io-touchpad/app/threads/application.py", line 116, in send_points_to_interpreter
    classifier.add_to_training_set(signal_list)
  File "/home/michal/io-project/io-touchpad/app/classifier/classifier.py", line 121, in add_to_training_set
    self.learn(False)
  File "/home/michal/io-project/io-touchpad/app/classifier/classifier.py", line 191, in learn
    .get_features(training_element))
  File "/home/michal/io-project/io-touchpad/app/classifier/featureextractor.py", line 378, in get_features
    angles = get_angle_list(new_curve)
  File "/home/michal/io-project/io-touchpad/app/classifier/featureextractor.py", line 294, in get_angle_list
    angle = angle_between_line_and_xaxis(point, next_point)
  File "/home/michal/io-project/io-touchpad/app/classifier/featureextractor.py", line 275, in angle_between_line_and_xaxis
    if point2.y != point1.y:
AttributeError: 'Point' object has no attribute 'y'

**Comment by piotrekp1** [comment]
i fixed it

File: app/test/test_classifier.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def test_learn():
     classifier.files[classifier_module.TRAINING_SET_FILE] = \
         "learn_dat/training-set.dat"
     classifier.learn(True)
-    assert fabs(classifier.tolerance_distance-1271.9887310656133650) < epsilon
+    assert fabs(classifier.tolerance_distance-1292.524372765934) < epsilon
```
Comment (0mp) [inline app/test/test_classifier.py]: What is this magic number?

**Comment by 0mp** [comment]
I think you can merge it.

**Pull request merged**